src/t8_forest/t8_forest_adapt.cxx

```diff
@@ -360,15 +401,17 @@ t8_forest_adapt (t8_forest_t forest)
                                                &el_inserted, elements);
           }
         }
-        el_considered += num_children;
+        el_considered += num_siblings;
       }
       else {
+
         /* The considered elements are neither to be coarsened nor is the first
          * one to be refined.
          * We copy the element to the new element array. */
         T8_ASSERT (refine == 0);
         elements[0] = t8_element_array_push (telements);
         tscheme->t8_element_copy (elements_from[0], elements[0]);
+        num_siblings = tscheme->t8_element_num_siblings (elements[0]);
```

Comment:
Is this recomputation really necessary?
I think we can remove this line.

Reply:
You are right, this line can be removed